src/lib/y2network/interface_config_builder.rb

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "yast"
+
+module Y2Network
+  # Collects data from the UI until we have enough of it to create a {Y2Network::Interface}.
+  # {LanItemsClass#Commit LanItems.Commit(builder)} uses it.
+  class InterfaceConfigBuilder
+    # @return [String] Device name (eth0, wlan0, etc.)
+    attr_accessor :name
+    # @return [String] type which is intended to be build
+    attr_accessor :type
+
+    # Constructor
+    #
+    # Load with reasonable defaults;
+    # see LanItems::new_item_default_options, LanItems::@SysconfigDefaults and
+    # others as in LanItems::Select
+    def initialize
+      @config = init_device_config({})
+      @s390_config = init_device_s390_config({})
+    end
+
+    # Stores one option value tuple
+    # FIXME: obsolete by []=
+    def set(option: nil, value: nil)
+      @config[option] = value
+    end
+
+    # Returns currently stored option value
+    # FIXME: obsolete by []
```

Comment:
Should we add a comment here about why the new code keeps using this method even if it's obsolete?

Reply:
bcs I was lazy to write a regex to replace it ;-) - fixed